proxy/http/HttpSM.cc

```diff
@@ -295,17 +297,9 @@ HttpSM::init()
   milestones[TS_MILESTONE_SM_START] = Thread::get_hrtime();
 
   magic = HTTP_SM_MAGIC_ALIVE;
-  sm_id = 0;
-
-  // Unique state machine identifier.
-  //  changed next_sm_id from int64_t to int because
-  //  atomic(32) is faster than atomic64.  The id is just
-  //  for debugging, so it's OK if it wraps every few days,
-  //  as long as the http_info bucket hash still works.
-  //  (To test this, initialize next_sm_id to 0x7ffffff0)
-  //  Leaving sm_id as int64_t to minimize code changes.
 
-  sm_id                    = (int64_t)ink_atomic_increment((&next_sm_id), 1);
+  // Unique state machine identifier
+  sm_id                    = next_sm_id++;
   t_state.state_machine_id = sm_id;
```

Comment:
Why do we even need t_state.state_machine_id ? Where it's used (Debug() ), we could just do state_machine->sm_id instead ?